cupyx/scipy/interpolate/_bspline.py

```diff
@@ -0,0 +1,912 @@
+
+import operator
+from itertools import product
+
+import cupy
+from cupy._core import internal
+from cupy._core._scalar import get_typename
+
+from cupyx.scipy.sparse import csr_matrix
+
+import numpy as np
+
+TYPES = ['float', 'double', 'complex<float>', 'complex<double>']
+INT_TYPES = ['int', 'long long']
+
+INTERVAL_KERNEL = r'''
+#include <cupy/complex.cuh>
+
+template<typename T>
+__global__ void find_interval(
+        const T* t, const T* x, long long* out,
+        int k, int n, bool extrapolate) {
```

Comment:
For my education: this is a C++ bool, correct? What's the story of converting C++ bools to python bools here -- note that Cython has a special `bint` type to avoid a mess with C bools vs python bools.

Reply:
We rely on `numpy.bool_` to get C bools from Python bools.
https://github.com/cupy/cupy/blob/6b31424246eb7d0e27a330b075211921a24406ae/cupy/cuda/function.pyx#L141-L142